src/main/java/com/palantir/gradle/versions/GradleConfigurations.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.plugins.JavaPluginConvention;
+
+final class GradleConfigurations {
+    /**
+     * Filters out both the unresolvable configurations but also the legacy java configurations that should not be
+     * resolved.
+     */
+    public static Stream<Configuration> getResolvableConfigurations(Project project) {
+        return project.getConfigurations().stream()
+                .filter(Configuration::isCanBeResolved)
+                .filter(conf -> !getLegacyJavaConfigurations(project).contains(conf.getName()));
```

Comment:
Should we pull this set out to a local variable captured by lambda to avoid computing it for each configuration?